tests/src/test/scala/scalaz/ImmutableArrayTest.scala

```diff
@@ -5,10 +5,13 @@ import std.anyVal._
 
 object ImmutableArrayTest extends SpecLite {
 
+  /*
+   TODO: Talk about this.
```

Comment:
I'm not sure what happened here, but it seems like using `B >~> A` instead of `B >: A` caused some type inference issues.  This test no longer compiles because 

```
[error] /Users/jacobbarber/projects/scalaz/tests/src/test/scala/scalaz/ImmutableArrayTest.scala:9: could not find implicit value for parameter ev: String >~> Int
[error]     val xs = ImmutableArray.fromArray(Array(1)) ++ ImmutableArray.fromArray(Array("a"))
```

This should be attempting to Lub in to `AnyRef`.